src/tpm2_asn.c

```diff
@@ -157,7 +157,7 @@ int TPM2_ASN_DecodeX509Cert(uint8_t* input, int inputSz,
     }
 
     /* Store certificate location */
-    if (rc == 0) {
+    if (rc >= 0) {
```

Comment:
Why not just change `TPM2_ASN_DecodeTag`, `TPM2_ASN_GetHeader` and `TPM2_ASN_GetLength_ex` to return 0 on success. The length is already returned in arguments...? I would much prefer fixing the return code of these functions then refactor things for >= and not even using rc.

Reply:
Tesfa let me know its because we are reusing the code from asn.c as much as possible. Allowing this.